File: TournamentOrganizer/TournamentOrganizer.DAL/Entities/Round.cs
using System;
using System.Collections.Generic;
using System.Linq;
using System.Text;
using System.Threading.Tasks;

namespace TournamentOrganizer.DAL.Entities
{
    public class Round
    {
        public Guid Id { get; set; }
        public int RoundNumber { get; set; }

        // Foreign Key
        public Guid TournamentId { get; set; }

        // Navigation Properties
        public Tournament Tournament { get; set; } = null!;
        public ICollection<Match> Matches { get; set; } = new List<Match>();
    }
}

File: tournamentorganizer-react/types/round.ts
import { Match } from "./match";
import { MatchOverview } from "./match";

export interface Round {
  id: string;
  roundNumber: number;
  matches: Match[];
}
export interface RoundOverview {
  id: string;
  roundNumber: number;
  matches: MatchOverview[];
}
